src/managers/PointerManager.cpp

```diff
@@ -668,68 +668,71 @@ void CPointerManager::move(const Vector2D& deltaLogical) {
 }
 
 void CPointerManager::warpAbsolute(Vector2D abs, SP<IHID> dev) {
-
-    PHLMONITOR currentMonitor = g_pCompositor->m_pLastMonitor.lock();
-    if (!currentMonitor || !dev)
+    if (!dev)
         return;
 
     if (!std::isnan(abs.x))
         abs.x = std::clamp(abs.x, 0.0, 1.0);
     if (!std::isnan(abs.y))
         abs.y = std::clamp(abs.y, 0.0, 1.0);
 
-    // in logical global
-    CBox mappedArea = currentMonitor->logicalBox();
+    // find x and y size of the entire space
+    Vector2D bottomRight = {-9999999, -9999999}, topLeft = {9999999, 9999999};
+    for (auto const& m : g_pCompositor->m_vMonitors) {
+        const auto EXTENT = m->logicalBox().extent();
+        const auto POS    = m->logicalBox().pos();
+        if (EXTENT.x > bottomRight.x)
+            bottomRight.x = EXTENT.x;
+        if (EXTENT.y > bottomRight.y)
+            bottomRight.y = EXTENT.y;
+        if (POS.x < topLeft.x)
+            topLeft.x = POS.x;
+        if (POS.y < topLeft.y)
+            topLeft.y = POS.y;
+    }
+    CBox mappedArea = {topLeft, bottomRight - topLeft};
 
     switch (dev->getType()) {
         case HID_TYPE_TABLET: {
             CTablet* TAB = reinterpret_cast<CTablet*>(dev.get());
             if (!TAB->boundOutput.empty()) {
                 if (const auto PMONITOR = g_pCompositor->getMonitorFromString(TAB->boundOutput); PMONITOR) {
-                    currentMonitor = PMONITOR->self.lock();
-                    mappedArea     = currentMonitor->logicalBox();
+                    if (const auto boundMonitor = PMONITOR->self.lock(); boundMonitor)
+                        mappedArea = boundMonitor->logicalBox();
+
+                    // apply the bound box relative to the bound output
+                    mappedArea.translate(TAB->boundBox.pos());
+                    if (!TAB->boundBox.empty()) {
+                        mappedArea.w = TAB->boundBox.w;
+                        mappedArea.h = TAB->boundBox.h;
+                    }
+                }
+            } else {
+                mappedArea.x = TAB->boundBox.x;
+                mappedArea.y = TAB->boundBox.y;
```

Comment:
This causes the origin to move to (0, 0) and make monitor regions in negative coordinates deadzones.

Reply:
oh yeah forgot about that. I'd have to add another option like `bool custom_region_position` or just always treat the region as relative.

Do you think it's worth it? I'd like being able to set an absolute tablet region because it makes mapping a tablet to a window really easy by just using the window's position and size from `hyprctl clients`